schemas/2018-04-01/Microsoft.Devices.json

```diff
@@ -0,0 +1,1044 @@
+          "description": "The Resource type.",
+          "enum": [
+            "Microsoft.Devices/IotHubs"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2018-04-01"
+          ]
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location."
+        },
+        "tags": {
+          "oneOf": [
+            {
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              }
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "The resource tags."
+        },
+        "etag": {
+          "type": "string",
+          "description": "The Etag field is *not* required. If it is provided in the response body, it must also be provided as a header per the normal ETag convention."
+        },
+        "properties": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/IotHubProperties"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "IotHub properties"
+        },
+        "sku": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/IotHubSkuInfo"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "IotHub SKU info"
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "oneOf": [
+              {
+                "$ref": "#/definitions/IotHubs_certificates_childResource"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "name",
+        "type",
+        "apiVersion",
+        "location",
+        "properties",
+        "sku"
+      ],
+      "description": "Microsoft.Devices/IotHubs"
+    },
+    "IotHubs_certificates": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "oneOf": [
+            {
+              "type": "string",
+              "pattern": "^[A-Za-z0-9-._]{1,64}$"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "The name of the certificate"
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Devices/IotHubs/certificates"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2018-04-01"
+          ]
+        },
+        "certificate": {
```

Comment:
>certificate [](start = 9, length = 11)

this is not allowed. al the RP specific properties should go into "properties" bag.

Reply:
We did get an exception from Gaurav on Certificates. Should I remove the Certificates from schema?